tanner/api.py

```diff
@@ -38,3 +100,52 @@ def __init__(self):
             for (i, val) in enumerate(query_res):
                 query_res[i] = json.loads(val)
         return query_res
+
+    async def return_session_info(self, sess_uuid, snare_uuid= None):
+        query_res = []
+        if snare_uuid:
+            snare_uuids = [snare_uuid]
+        else:
+            snare_uuids = await self.return_snares()
+
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            if sessions == 'Invalid SNARE UUID':
+                continue
+            for sess in sessions:
+                if sess['sess_uuid'] == sess_uuid:
+                    return sess
+
+    async def return_sessions(self, filters):
+        query_res = []
+        if 'snare_uuid' in filters:
+            snare_uuids = [filters['snare_uuid']]
+        else:
+            snare_uuids = await self.return_snares()
+
+        matching_sessions = []
+        for snare_id in snare_uuids:
+            sessions = await self.return_snare_info(snare_id)
+            for sess in sessions:
+                if sessions == 'Invalid SNARE UUID':
+                    continue
+                is_matching_sesssion = True
+                if 'user-agent' in filters:
```

Comment:
I suggest you to use more general approach here
e.g.
for every `key, value` in `filters` return session which `sess[key]==value`
and use `try` `except` for KeyError